src/signavault/api.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * @packageDocumentation
+ * @module API-multisig
+ */
+import { AxiosError, AxiosRequestConfig } from "axios"
+import { StoreAPI } from "store2"
+import Signavault from "caminojs/signavault/index"
+import DB from "caminojs/utils/db"
+import { RequestResponseData } from "caminojs/common"
+
+export type Owner = {
+  address: string
+  signature: string
+}
+export type MultisigTx = {
+  id: string
+  unsignedTx: string
+  alias: string
+  threshold: number
+  transactionId: string
+  owners: Owner[]
+  outputOwners: string
+}
+export type TxID = {
+  txID: string
+}
+
+export class MultisigAPI {
+  protected db: StoreAPI
+  protected baseURL: string
+  protected core: Signavault
+
+  setBaseURL = (baseURL: string) => {
+    if (this.db && this.baseURL !== baseURL) {
+      const backup = this.db.getAll()
+      this.db.clearAll()
+      this.baseURL = baseURL
+      this.db = DB.getNamespace(baseURL)
+      this.db.setAll(backup, true)
+    } else {
+      this.baseURL = baseURL
+      this.db = DB.getNamespace(baseURL)
+    }
+  }
+
+  getBaseURL = (): string => this.baseURL
+  getDB = (): StoreAPI => this.db
+
+  createTx = async (
+    alias: string,
+    unsignedTx: string,
+    signature: string,
+    outputOwners: string,
+    headers?: object
+  ): Promise<MultisigTx> => {
+    const endpoint = ""
+    const body: any = {}
+    body.alias = alias
+    body.unsignedTx = unsignedTx
+    body.signature = signature
+    body.outputOwners = outputOwners
+
+    let headrs: object = { "Content-Type": "application/json;charset=UTF-8" }
+    if (headers) {
+      headrs = { ...headrs, ...headers }
+    }
+
+    const axConf: AxiosRequestConfig = {
+      baseURL: this.core.getURL(),
+      responseType: "json"
+    }
+
+    var resp: RequestResponseData
+    try {
+      resp = await this.core.post(
+        this.baseURL + endpoint,
+        {},
+        JSON.stringify(body),
+        headrs,
+        axConf
+      )
+    } catch (e: any) {
+      throw e.message ?? (e as AxiosError).toJSON()
+    }
+
+    if (resp.status >= 200 && resp.status < 300) {
+      return resp.data
+    } else {
+      throw new Error(resp.data.message + " error: " + resp.data.error)
+    }
+  }
```

Comment:
The only thing that changes here are the HTTP methods and the body object, which is generic. Is there a clean way to facor this into a utility call?